R/HintikkaXOData.R

```diff
@@ -35,42 +33,61 @@
 #' \emph{International Journal of Environmental Research and Public Health}
 #' 18(8):4049 \url{https://doi.org/10.3390/ijerph18084049}
 #'
+#' @importFrom SummarizedExperiment rowData
+#' @importFrom SummarizedExperiment rowData<-
+#' 
 #' @name HintikkaXOData
-#' @importFrom SingleCellExperiment altExp
-#' @importFrom SingleCellExperiment altExp<-
 #' @export
 #'
 #' @examples
-#' # tse <- HintikkaXOData()
+#' # Retrieve the MAE data
+#' mae <- HintikkaXOData()
+#'
+#' # List the experiments in this MultiAssayExperiment object
+#' print(experiments(mae))
+#'
+#' # colData for this MAE data object
+#' colData(mae)
+#'
+#' # metabolite assay data
+#' # assays(mae[["metabolites"]])$assay
+#'
+#' # biomarker assay data
+#' # assays(mae[["biomarkers"]])$assay
+# '
+#' # microbiome assay counts
+#' # assays(mae[["microbiome"]])$counts
 #'
+#' # microbiome rowData
+#' rowData(mae[["microbiome"]])
+#' 
 HintikkaXOData <- function() {
 
-    # Circumvent warnings
-    meta_cecum <- otu_cecum <- tax <- NULL
-
-    dataset <- "3.14/hintikka-xo"
-    se <- .create_se(dataset,
-                    assays = c("counts"),
-                    has.rowdata = TRUE,
-                    has.coldata = TRUE)
-
-    tse <- SummarizedExperiment(assays = list(counts = otu_cecum),
-                        colData = meta_cecum,
-                        rowData = tax)
+    mae <- .create_mae("3.14/hintikka-xo",
+    
+        types = list(microbiome  = list("TSE" = c("counts")),
+                    metabolites = list("SE"  = NULL),
+                    biomarkers  = list("SE"  = NULL)
+                ),
+ 
+        coldata = TRUE,
+        samplemap = FALSE,
 
-    # Create SE objects
-    bm  <- SummarizedExperiment(bm)
-    nmr <- SummarizedExperiment(nmr)
+        has.rowdata = list(microbiome = TRUE,
+                        metabolites = FALSE,
+                        biomarkers  = FALSE),
 
-    # Add Biomarkers as "alternative experiment":
-    altExp(se, "Biomarkers") <- bm
+        has.coldata = list(microbiome = FALSE,
+                        metabolites = FALSE,
+                        biomarkers  = FALSE))
 
-    # Add NMR metabolite abundances as "alternative experiment":
-    altExp(se, "NMR") <- nmr
 
-    se
+    # Fix order in rowData
+    rowData(mae[["microbiome"]]) <- rowData(mae[["microbiome"]])[, c(2:7, 1)]
```

Comment:
Yes I know. But now as the data is already in ExperimentHub, I was planning to possibly update this when we send some other new data there as well because it seems to require action from them at every update. Or is it sufficient that I just push data updates to the bucket?